graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DataTypeMapping.java

```diff
@@ -70,47 +135,62 @@ static Object toDbValue(Column.ColumnType type, Object value) {
     Map<String, Object> map = new HashMap<>(columns.size());
     for (Column column : columns) {
       if (!row.isNull(column.name())) {
-        map.put(nameMapping.getColumnName(table).get(column), toGraphQLValue(column, row));
+        map.put(
+            nameMapping.getColumnNames(table).get(column),
+            toGraphQLValue(nameMapping, column, row));
       }
     }
     return map;
   }
 
-  private static Object toGraphQLValue(Column column, Row row) {
+  private static Object toGraphQLValue(NameMapping nameMapping, Column column, Row row) {
     Object dbValue = row.getObject(column.name());
-    return toGraphQLValue(column.type(), dbValue);
+    return toGraphQLValue(nameMapping, column.type(), dbValue);
   }
 
-  private static Object toGraphQLValue(Column.ColumnType type, Object dbValue) {
-    if (!type.isCollection() || dbValue == null) {
+  private static Object toGraphQLValue(
+      NameMapping nameMapping, Column.ColumnType type, Object dbValue) {
+    if (dbValue == null) {
+      return null;
+    } else if (type.isCollection()) {
+      if (type.rawType() == Column.Type.List || type.rawType() == Column.Type.Set) {
+        Collection<?> dbCollection = (Collection<?>) dbValue;
+        return dbCollection.stream()
+            .map(item -> toGraphQLValue(nameMapping, type.parameters().get(0), item))
+            .collect(Collectors.toList());
+      } else if (type.rawType() == Column.Type.Map) {
+        Map<?, ?> dbMap = (Map<?, ?>) dbValue;
+        Column.ColumnType keyType = type.parameters().get(0);
+        Column.ColumnType valueType = type.parameters().get(1);
+        return dbMap.entrySet().stream()
+            .map(
+                e -> {
+                  Map<String, Object> m = new HashMap<>(2);
+                  m.put("key", toGraphQLValue(nameMapping, keyType, e.getKey()));
+                  m.put("value", toGraphQLValue(nameMapping, valueType, e.getValue()));
+                  return m;
+                })
+            .collect(Collectors.toList());
+      } else {
+        throw new AssertionError("Invalid collection type " + type);
+      }
+    } else if (type.isUserDefined()) {
+      UserDefinedType udt = (UserDefinedType) type.frozen(false);
+      UdtValue udtValue = (UdtValue) dbValue;
+      Map<String, Object> result = new HashMap<>(udt.columns().size());
+      for (Column column : udt.columns()) {
+        Object dbFieldValue = udtValue.getObject(CqlIdentifier.fromInternal(column.name()));
+        if (dbFieldValue != null) {
```

Comment:
Both UDTs and GraphQL fields can validly contain null values. Do we want to return the null here?

Reply:
nm, this work fine.